### ISSUE TITLE
manual_backward

### ISSUE THREAD

**Issue by 7zhanghao** [PyTorchLightning/pytorch-lightning#12811]
## 🐛 Bug

When i want to manually manage the optimization process, the gradient accumulation need to do by myself. The example you give is:
`opt = self.optimizers()`
`loss = self.compute_loss(batch)`
`self.manual_backward(loss)`
`accumulate gradients of 'n' batches`
`if (batch_idx + 1) % n == 0:`
    `opt.step()`
    `opt.zero_grad()`
Because the gradients are accumulated n times, we need to take the average and divide by n. This is usually achieved by dividing each loss by 4. However, this is not done by the exapmle. I think the right usage is `loss = self.compute_loss(batch) / n`. Is there any problem?
### Environment

- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @borda @rohitgr7

**Comment by rohitgr7** [comment]
yes, with manual backward you have to do it. In the case of automatic optimization, it is done by default
https://github.com/PyTorchLightning/pytorch-lightning/blob/908e05880d4271ad32876311320d4465a008a710/pytorch_lightning/loops/optimization/optimizer_loop.py#L84-L87

**Comment by 7zhanghao** [comment]
> yes, with manual backward you have to do it. In the case of automatic optimization, it is done by default
> 
> https://github.com/PyTorchLightning/pytorch-lightning/blob/908e05880d4271ad32876311320d4465a008a710/pytorch_lightning/loops/optimization/optimizer_loop.py#L84-L87

Thanks reply. Another problem is that I am using ddp mode. Under this circumstances, if i manually manage the optimization process with `self.manual_backward(loss)`, can ddp work correctly when using ddp? If so, can ddp work correctly if i use `loss.backward()` instead of `self.manual_backward(loss)`?

**Comment by 7zhanghao** [comment]
> ```python
> self.manual_backward(loss)
> ```
> 
> takes care of scaling the loss when using amp and hook calls. Is there any issue with using `self.manual_backward`?

No more questions, thanks a lot.

**Comment by callumhay** [comment]
I think the documentation should be updated to reflect that the loss for manual optimization with gradient accumulation requires manual scaling with the accumulation factor 'N'. The [current doc](https://lightning.ai/docs/pytorch/stable/model/manual_optimization.html) is a bit confusing since it specifies the use of % N for the optimizer stepping and zeroing of gradients but implicitly says nothing about scaling the actual loss and instead leaves it to the reader to scrutinize what `self.compute_loss(batch)` actually does.

I think it would be more instructive to have the example code specifically call this out, e.g., `loss = self.my_loss_fn(batch) / N`.
